firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -0,0 +1,262 @@
+
+    WHEN("The output method is called")
+      THEN("The output method returns waiting status")
+      THEN("The output buffer remains unchanged")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("the input method reports output ready status on the delimeter byte")
+      THEN("the output method reports ok status")
+      THEN("The output buffer is as expected")
+
+  GIVEN("A uint8_t chunksplitter constructed with value of delimeter parameter as 0xff is completely filled with non-delimeter input data")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("257th non-delimiter byte is passed as input")
+      THEN("The input method reports invalid_length status")
+
+    WHEN("The output method is called")
+      THEN("The output method returns waiting status")
+      THEN("The output buffer remains unchanged")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("the input method reports output ready status on the delimeter byte")
+      THEN("the output method reports ok status")
+      THEN("The output buffer is as expected")
+
+Scenario: A chunksplitter constructed with a non-zero delimeter parameter and include_delimeter boolean as true correctly handles input chunks
+
+  GIVEN("A uint8_t chunksplitter with a completely full internal buffer of capacity 256 bytes")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("257th non-delimiter byte is passed as input")
+      THEN("The input method reports invalid_length status")
+
+    WHEN("The output is called after the internal buffer is partially filled")
+      THEN("The output method reports waiting status")
+      THEN("The output buffer remains unchanged")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("the input method reports invalid_length status on the delimeter byte")
+      THEN("the output method reports invalid length status")
+      THEN("The output buffer is as expected")
+
+  GIVEN("A uint8_t chunksplitter with a partially filled internal buffer of capacity 256 bytes")
+  // assertions:
+  // output buffer is empty  
+
+    WHEN("The output is called after the internal buffer is partially filled")
+      THEN("The output method reports waiting status")
+      THEN("The output buffer remains unchanged")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("The input method reports output_ready status on the delimeter byte")
+      THEN("the output method reports ok status")
+      THEN("The output buffer is as expected")
+
+Scenario: A ChunkSplitter constructed with default parameters correctly handles input of multiple delimited chunks before output consumes them
+
+  GIVEN("A uint8_t chunksplitter with an empty internal buffer of capacity 256 bytes")
+
+    WHEN("input to the chunksplitter is '0x01 0x02 0x03 0x00 0x04 0x05 0x00' and output is called after each delimeter input")
```

Comment:
Output is called after each delimiter, so we should also have THENs about the outputs of the first output call, not just the last one. Since the next WHEN includes THENs about input_overwritten, and since this scenario is about input_overwritten, I think this WHEN should also have a THEN about input_overwritten after each input method call, as it should always be left unchanged (unless I've forgotten something). See my next comment about testing for whether input_overwritten is reset to false or is left unchanged.